hydromt/models/root.py

```diff
@@ -58,6 +86,30 @@ def is_reading_mode(self) -> bool:
         """Test whether we are in reading mode or not."""
         return self._mode.is_reading_mode()
 
+    # old_root is so we can copy over the old log file if needed.
+    def _create_loggers(self, old_path: Optional[Path] = None, overwrite: bool = False):
+        if not exists(self._path):
+            mkdir(self._path)
+
+        log_level = 20  # default, but overwritten by the level of active loggers
+        for i, h in enumerate(self.logger.handlers):
+            log_level = h.level
+            if isinstance(h, FileHandler):
+                if dirname(h.baseFilename) != self._path:
```

Comment:
In case we are already logging to `self._path` there is no need to create a new filehandler right? Otherwise we would be logging the same message twice to the file. If the logfile already exists in the handlers, I think we can exit the method?

Reply:
We could exit the loop instead, but if we exit the method we won't overwrite the logs if you just change the mode to overwrite which was part of the functional requirements.

Reply:
I added an early exit if the old and new paths are the same since then we don't have to do anything, (and from your comment I infer that in that case we don't have to overwrite the old log file)